src/encoded/static/components/schema.js

```diff
@@ -89,32 +94,143 @@ function createJsonDisplay(elementId, term) {
 
 
 /**
- * Convert the contents of the `id` field of a schema to a path we can use to link to the page to
- * add a new object of its type.
+ * Convert the contents of the `id` field of a schema to a schema name.
  *
  * @param {*} schemaId - `id` property from schema
  */
-function schemaIdToPath(schemaId) {
+function schemaIdToName(schemaId) {
     const pathMatch = schemaId.match(/\/profiles\/(.*).json/);
     return pathMatch && pathMatch.length ? pathMatch[1] : null;
 }
 
 
+/**
+ * Convert the schema id property value to a page URI.
+ *
+ * @param {string} schemaId - value of id property within a schema
+ */
+function schemaIdToPage(schemaId) {
+    return schemaId.slice(0, -5);
+}
+
+
+/**
+ * Collect any linkTo and linkFrom properties within the schema object in the `prop` object. Two
+ * arrays of these string properties get returned as an object with this format...
+ *
+ * {
+ *     linkTo - array of linkTo property strings
+ *     linkFrom - array of linkFrom property strings
+ * }
+ *
+ * @param {object} prop - Schema property containing its own terms.
+ */
+function collectLinks(prop) {
+    const propKeys = Object.keys(prop);
+    const linksLists = { linkTo: [], linkFrom: [] };
+
+    propKeys.forEach((key) => {
+        if (key === 'linkTo' || key === 'linkFrom') {
+            // Simple linkTo or linkFrom property; adds its value to the array we're collecting.
+            linksLists[key].push(key === 'linkFrom' ? prop[key].split('.')[0] : prop[key]);
+        } else if (typeof prop[key] === 'object' && !Array.isArray(prop[key])) {
+            // The schema property key is not one of `nonlinkProps` and has a value that's an object.
```

Comment:
I don't see `nonLinkProps` anywhere except in this comment.

Reply:
Yeah, nonLinkProps used to be there but I changed it.